floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,384 @@
+    """
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {api_token}",
+        "nulls": "0"
+    }
+    allPlayersFromActivity_response = _getAllplayerFormActivity_response(baseUrl, api_token, activity_id)
+    allPlayersFromActivity_JSON = allPlayersFromActivity_response.json()
+    playerIDsInActivity_list = list(set(i["id"] for i in allPlayersFromActivity_JSON))
+    response_data_json_list = []
+
+    for i in playerIDsInActivity_list:
+        url = f'activities/{activity_id}/athletes/{i}/sensor'
+        response_data_json_i = requests.get(baseUrl + url, headers=headers)
+        if response_data_json_i.status_code == 200:
+            response_data_json_list.append(response_data_json_i.json()[0])
+            saveResponse(response_data_json_i, url)
+        else:
+            raise APIRequestError(f"API request failed with status code {response_data_json_i.status_code}")
+
+    return response_data_json_list
+
+def get_key_names_from_json(response_data_json_list: List[Dict]) -> set:
+    """
+    Extracts unique column names from a list of JSON data responses.
+
+    Args:
+        response_data_json_list (List[Dict]): List of JSON data responses.
+
+    Returns:
+        set: A set of unique column names.
+    """
+    recorded_columns = set()
+
+    for data_i in response_data_json_list:
+        recorded_columns.update(data_i.keys())
+        if "data" in data_i and isinstance(data_i["data"], list) and data_i["data"]:
+            recorded_columns.update(data_i["data"][0].keys())
+
+    return recorded_columns
+
+def _get_key_links(response_data_json_list: List[Dict]) -> Union[Dict[str, int], None]:
+    """
+    Generates a mapping of key names to their indices in the JSON data responses.
+
+    Args:
+        response_data_json_list (List[Dict]): List of JSON data responses.
+
+    Returns:
+        Union[Dict[str, int], None]: A dictionary mapping key names to their indices or None if critical keys are missing.
+    """
+    for data_i in response_data_json_list:
+        data_i['name'] = f"{data_i['athlete_first_name']} {data_i['athlete_last_name']}"
+
+    key_names = list(get_key_names_from_json(response_data_json_list))
+
+    mapping = {
+        "ts": "time",
+        "device id": "sensor_id",
+        "athlete_id": "mapped_id",
+        "athlete_first_name": "athlete_first_name",
+        "athlete_last_name": "athlete_last_name",
+        "stream_type": "stream_type",
+        "name": "name",
+        "jersey": "number",
+        "team_id": "group_id",
+        "team_name": "group_name",
+        "x": "x_coord",
+        "y": "y_coord",
+        "cs": "cs",
+        "lat": "lat",
+        "long": "long",
+        "o": "o",
+        "v": "v",
+        "a": "a",
+        "hr": "hr",
+        "pl": "pl",
+        "mp": "mp",
+        "sl": "sl",
+    }
+
+    necessary_keys = ["time", "x_coord", "y_coord"]

```

Comment:
I think another option for `necessary_keys` should be `["time", "lat", "long"]`. So in the check for critical information two types of necessary_keys will be tested. Although I doubt it is possible to get a the API to return one without another but I'd rather have this explicitly here.

Reply:
Done